samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -36,6 +36,7 @@
   private final Object key;
   private final Object message;
   private final int size;
+  private long timestamp = 0L;
```

Comment:
Can you add context (maybe in the JIRA) for why this is required?
Also, can we set this in the constructor for every IME instead of a setter called by specific consumers?

Reply:
Added JIRA.
I didn't want neither change the existing constructor nor add another one to IME. So I thought it is easier just to add getter/setter.

Reply:
Apologies for the late reply. I was suggesting set it as System.currentTimeMillis() by default, either in the IME constructors or in the field declaration so that users get a reasonable value when they use it.

+1 for Javadocs. Since this is a public API, we should document what this timestamp represents.

Reply:
@sborya Is the task to add documentation + default timestamp for the IME tracked somewhere? We can try to finish it before the next major release.